From a95a56862efa4b2d15ea24c4cdb635730c12da41 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=D0=BD=D0=B0=D0=B1?= <nabijaczleweli@nabijaczleweli.xyz>
Date: Thu, 28 Jul 2022 20:09:38 +0200
Subject: [PATCH 5/5] Write first chunk of SSH_CLIENT to ut_host
X-Mutt-PGP: OS

---
 posix.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/posix.c b/posix.c
index 791d618..2040aa6 100644
--- a/posix.c
+++ b/posix.c
@@ -97,6 +97,14 @@ addutmp(void)
 	clock_gettime(CLOCK_REALTIME, &now);
 	utmp.ut_tv.tv_sec = now.tv_sec;
 	utmp.ut_tv.tv_usec = now.tv_nsec / 1000;
+	if((pts = getenv("SSH_CLIENT"))) {
+		cp = strchr(pts, ' ');
+		if(cp)
+			*cp = '\0';
+		strncpy(utmp.ut_host, pts, sizeof(utmp.ut_host));
+		if(cp)
+			*cp = ' ';
+	}
 #if __GLIBC__
 	utmp.ut_session = getsid(0);
 #endif
-- 
2.30.2

